vllm/model_executor/layers/rejection_sampler.py

```diff
@@ -250,12 +263,21 @@ def _smallest_positive_value(self) -> float:
 def _multinomial(
     probs: torch.Tensor,
     num_samples: int,
+    generators: List[Optional[torch.Generator]],
 ) -> torch.Tensor:
+
     if num_samples > 1:
         # This is equivalent to torch.repeat_interleaved (which also
         # forces a GPU<->CPU sync).
         probs = probs[:, None, :].expand(probs.shape[0], num_samples,
                                          probs.shape[1]).contiguous().view(
                                              -1, probs.shape[1])
-    q = torch.empty_like(probs).exponential_(1.0)
+
+    q = torch.empty_like(probs)
+    if any(generator is not None for generator in generators):
+        for i, generator in enumerate(generators):
+            q[i].exponential_(1.0, generator=generator)
```

Comment:
Similar comment to above about doing single `exponential_` for all non-seeded rows

Reply:
done